src/main/clm_initializeMod.F90

```diff
@@ -478,40 +482,66 @@ subroutine initialize2(ni,nj)
+                write(iulog,'(a)')'Using cold start initial conditions '
              end if
           else
              if (masterproc) then
-                write(iulog,*)'Interpolating initial conditions from ',trim(finidat_interp_source),&
-                     ' and creating new initial conditions ', trim(finidat_interp_dest)
+                write(iulog,'(a)')'Interpolating initial conditions from '//trim(finidat_interp_source)
+                write(iulog,'(a)')'Creating new initial conditions file '//trim(finidat_interp_dest)
              end if
           end if
        else
+          if (trim(finidat) == trim(finidat_interp_dest)) then
+             ! Check to see if status file for finidat exists
+             klen = len_trim(finidat_interp_dest) - 3 ! remove the .nc
+             locfn = finidat_interp_dest(1:klen)//'.status'
+             inquire(file=trim(locfn), exist=lexists)
+             if (.not. lexists) then
+                if (masterproc) then
+                   write(iulog,'(a)')' failed to find file '//trim(locfn)
+                   write(iulog,'(a)')' this indicates a problem in creating '//trim(finidat_interp_dest)
+                   write(iulog,'(a)')' remove '//trim(finidat_interp_dest)//' and try again'
+                end if
+                call endrun()
+             end if
+          end if
           if (masterproc) then
-             write(iulog,*)'Reading initial conditions from ',trim(finidat)
+             write(iulog,'(a)')'Reading initial conditions from file '//trim(finidat)
           end if
           call getfil( finidat, fnamer, 0 )
           call restFile_read(bounds_proc, fnamer, glc_behavior, &
                reset_dynbal_baselines_lake_columns = reset_dynbal_baselines_lake_columns)
        end if
     else if ((nsrest == nsrContinue) .or. (nsrest == nsrBranch)) then
        if (masterproc) then
-          write(iulog,*)'Reading restart file ',trim(fnamer)
+          write(iulog,'(a)')'Reading restart file '//trim(fnamer)
        end if
        call restFile_read(bounds_proc, fnamer, glc_behavior, &
             reset_dynbal_baselines_lake_columns = reset_dynbal_baselines_lake_columns)
     end if
 
     ! If appropriate, create interpolated initial conditions
     if (nsrest == nsrStartup .and. finidat_interp_source /= ' ') then
-       is_interpolated_start = .true.
 
        ! Check that finidat is not cold start - abort if it is
        if (finidat /= ' ') then
           call endrun(msg='ERROR clm_initializeMod: '//&
                'finidat and finidat_interp_source cannot both be non-blank')
        end if
 
+       ! Determine name if finidat_interp_dest status file
+       klen = len_trim(finidat_interp_dest) - 3 ! remove the .nc
+       locfn = finidat_interp_dest(1:klen)//'.status'
+
+       ! Remove file if it already exists
+       inquire(file=trim(locfn), exist=lexists)
+       if (lexists) then
+          open(unit=1234, file=locfn, status='old', iostat=ioe)
```

Comment:
- [ ] Here and two lines down: this should use newunit rather than a hard-coded unit number that could theoretically already be in use.

Reply:
@billsacks - this does not work with newunit. I tried that. 

https://github.com/ESCOMP/CTSM/blob/2671dfc54aa91cc5c3ba5e3ca984046dfe1216a7/src/main/clm_initializeMod.F90#L539-L542

When you use newunit - the actual unit number is negative. And trying to use the delete flag on a negative unit number was problematic. @jedwards4b suggested this code and I think this is why he used 1234. I am happy to put a comment in here if that would be useful.

Reply:
Interesting, good to know. I guess you could use the mechanism we used before the existence of newunit, which is to use shr_file_getUnit and shr_file_freeUnit, but I'm okay with this being left as is. (I'd support adding a comment, but don't feel it's critical.)